packages/@lwc/engine-core/src/framework/modules/attrs.ts

```diff
@@ -63,3 +63,22 @@ export function patchAttributes(
         }
     }
 }
+
+export function patchSlotAssignment(
+    oldVnode: VBaseElement | VStatic | null,
+    vnode: VBaseElement | VStatic,
+    renderer: RendererAPI
+) {
+    if (oldVnode?.slotAssignment === vnode.slotAssignment) {
+        return;
+    }
+
+    const { elm } = vnode;
+    const { setAttribute, removeAttribute } = renderer;
+
+    if (isUndefined(vnode.slotAssignment) || isNull(vnode.slotAssignment)) {
```

Comment:
Let's hoist `vnode.slotAssignment` as a variable to the top of the function instead of calling the getter repeatedly. These kinds of things might matter for perf-sensitive code like this.

Reply:
Suggestion from PR review, run tachometer tests to see performance before/after this change.

Reply:
I would go ahead and just hoist the `vnode.slotAssignment` variable to be safe. It's a pattern we use a lot in these `patch*` functions anyway.

Reply:
Updated in ca69cc248769c3854e9347f02204473a68bdfbd2!

I also ran the perf tests and I didn't see a regression with these changes.